### PR TITLE
try to fix the little bug when calling Close or Flush in async mode

### PR DESCRIPTION
**Pull request by youngsterxyf**
In the Async mode, calling the `func (bl *BeeLogger) Flush() ` or `func (bl *BeeLogger) Close()`, sometimes some log message may be discard, or the order of log may be wrong. Please see the example:

```go
package main

import (
	"github.com/astaxie/beego/logs"
)

var log *logs.BeeLogger

func init() {
	log = logs.NewLogger(10000)
	log.SetLogger("console", `{"level": 10}`)
	log.SetLevel(logs.LevelDebug)
	log.EnableFuncCallDepth(true)
	log.SetLogFuncCallDepth(3)
	log.Async()
}

func main() {
	log.Debug("1 log message")
	log.Debug("2 log message")
	log.Debug("3 log message")
	log.Debug("4 log message")
	log.Debug("5 log message")

	log.Close()
}
```

the output:

image

you can see that first three times  the second log has been discard

### CODE REVIEW

File: logs/log.go [hunk 1]
```diff
@@ -98,6 +98,8 @@ type BeeLogger struct {
 	loggerFuncCallDepth int
 	asynchronous        bool
 	msgChan             chan *logMsg
+	signalChan			chan string
+	wg					sync.WaitGroup
```
Comment (astaxie) [inline logs/log.go]: fmt

Comment (youngsterxyf) [inline logs/log.go]: @astaxie sorry

Comment (astaxie) [inline logs/log.go]: use the atom or liteide. auto fmt when save

Comment (youngsterxyf) [inline logs/log.go]: @astaxie OK

File: logs/log.go [hunk 2]
```diff
@@ -345,27 +370,16 @@ func (bl *BeeLogger) Trace(format string, v ...interface{}) {
...
-		break
-	}
-	for _, l := range bl.outputs {
-		l.Flush()
-		l.Destroy()
-	}
-	bl.outputs = nil
+	bl.signalChan <- "close"
+	bl.wg.Wait()
+	bl.wg.Add(1)
```
Comment (astaxie) [inline logs/log.go]: if we close the Log, do we still need to add?

**Comment by youngsterxyf** [comment]
@astaxie with golang tip version, `go vet ./...` will raise an error.

image

**Comment by astaxie** [comment]
@youngsterxyf it should be fixed in develop now. I like travis

**Pull request merged**